lib/pbench/test/unit/server/test_shell_cli.py

```diff
@@ -231,6 +232,24 @@ def wait_for_oidc_server(
         assert called[0]
         assert ret_val == 1
 
+        called = [False]
+
+        # Test when specifically OpenIDClient.NotConfigured is raised from
+        # wait_for_oidc_server
+        def wait_for_oidc_server(
+            server_config: PbenchServerConfig, logger: logging.Logger
+        ) -> str:
+            called[0] = True
+            raise OpenIDClient.NotConfigured
```

Comment:
This is almost identical to the code at lines 218-222...including the function name.  While I suppose it's fine to redeclare a new function for the same name, it seems like a bad idea.  However, rather than changing the name, since this is an inner function why not have it raise an exception from a variable assigned in the outer scope?  That way the first part of the test can raise one exception and the second part can raise another exception, but they can both use the same mock function (we get less code and we don't have to worry another name).

Reply:
That's a good point.